indico/modules/events/registration/templates/display/event_header.html

```diff
@@ -55,7 +55,9 @@
             <div class="participant-list-wrapper">
                 <ul class="participant-list">
                     {%- for participant in participants -%}
-                        <li>{{ participant.full_name }}</li>
+                    <li {% if participant.checked_in and participant.registration_form.publish_checkin_enabled %}class="checked-in"{% endif %}>
```

Comment:
An even nicer alternative would be this:

    <li class="{% if participant.checked_in and participant.registration_form.publish_checkin_enabled %}checked-in{% endif %}">

Like this syntax highlighting in most editors will be less messed up since it's inside the attribute value. But either way is fine.